src/tests/to-do-list.test.tsx

```diff
@@ -0,0 +1,203 @@
+import React from 'react';
+import { render, screen, cleanup, fireEvent } from "@testing-library/react";
+import "@testing-library/jest-dom/extend-expect";
+import user from '@testing-library/user-event';
+import { ToDoView } from '../view/to-do/to-do.view';
+
+afterEach(
+    () => {
+        cleanup();
+    }
+);
+
+describe("to-do-view", () => {
+
+    it('should save two ToDos', () => {
+        render(<ToDoView />);
+        const ToDoViewElement = screen.getByTestId('to-do-view-container');
+        expect(ToDoViewElement).toBeInTheDocument();
+
+        const AddButtonElement = screen.getByText(/add/i);
+        expect(AddButtonElement).toBeInTheDocument();
+        const AddInputElement = screen.getByTestId('input-component');
+        expect(AddInputElement).toBeInTheDocument();
+
+        user.type(AddInputElement, 'Test first ToDo');
+        user.click(AddButtonElement);
+
+        let ItemListElement = screen.getAllByTestId('item-list-container');
+        expect(ItemListElement.length).toBe(1);
+
+        user.type(AddInputElement, 'Test second ToDo');
+        user.click(AddButtonElement);
+
+        ItemListElement = screen.getAllByTestId('item-list-container');
+        expect(ItemListElement.length).toBe(2);
+    });
+
+
+    it('should at start save a ToDo and after delete it', () => {
+        render(<ToDoView />);
+        const ToDoViewElement = screen.getByTestId('to-do-view-container');
+        expect(ToDoViewElement).toBeInTheDocument();
+
+        const AddButtonElement = screen.getByText(/add/i);
+        expect(AddButtonElement).toBeInTheDocument();
+        const AddInputElement = screen.getByTestId('input-component');
+        expect(AddInputElement).toBeInTheDocument();
+
+        user.type(AddInputElement, 'Test first ToDo');
+        user.click(AddButtonElement);
+
+        let ItemListElement = screen.getAllByTestId('item-list-container');
+        expect(ItemListElement.length).toBe(1);
+
+        const DeleteSvgElement = screen.getByTestId('delete-svg-item-list');
+        expect(DeleteSvgElement).toBeInTheDocument();
+
+        user.click(DeleteSvgElement);
+        expect(screen.queryByTestId('item-list-container')).toBeFalsy();
+    });
+
+    it('should should change the svg not-done to the svg done', () => {
```

Comment:
I think a better description would be "should mark an item as done or not done".
The fact that there is an svg being changed is an implementation detail.

Reply:
Yes, yours wording is indeed clearer for the purpose of understanding the test. I will try to improve my summarising and explanatory skills